src/cmd_sym.ml

```diff
@@ -219,55 +206,93 @@ let run_file ~unsafe ~optimize (pc : unit Result.t Choice.t) filename =
+  let testcase = `El (tag ?atts "testcase", testcase >>| input) in
+  let dtd =
+    {|<!DOCTYPE testcase PUBLIC "+//IDN sosy-lab.org//DTD test-format testcase 1.1//EN" "https://sosy-lab.org/test-format/testcase-1.1.dtd">|}
+  in
+  Xmlm.output o (`Dtd (Some dtd));
+  Xmlm.output_tree Fun.id o testcase
 
-let stop_at_first_failure = true
+let write_testcase =
+  let cnt = ref 0 in
+  fun ~dir ~err testcase ->
+    incr cnt;
+    let name = Format.sprintf "testcase-%d.xml" !cnt in
+    let path = Filename.concat dir name in
+    let out_chan = open_out path in
+    Fun.protect
+      ~finally:(fun () -> close_out out_chan)
+      (fun () -> out_testcase ~dst:(`Channel out_chan) ~err testcase)
 
-let cmd profiling debug unsafe optimize workers files =
+let cmd profiling debug unsafe optimize workers no_stop_at_failure workspace
+  files =
   if profiling then Log.profiling_on := true;
   if debug then Log.debug_on := true;
-  let solver = Thread.Solver.create () in
+  mkdir_p_exn workspace;
   let pc = Choice.return (Ok ()) in
+  let solver = Solver.create () in
   let result = List.fold_left (run_file ~unsafe ~optimize) pc files in
   let thread : Thread.t = Thread.create () in
   let results = Choice.run_and_trap ~workers result thread in
   let failing =
     Seq.filter_map
       (fun (result, thread) ->
+        let pc = Thread.pc thread in
         Format.printf "PATH CONDITION:@.";
-        List.iter
-          (fun c -> print_endline (Encoding.Expression.to_string c))
-          (Thread.pc thread);
-        match result with
-        | Choice_monad_intf.EVal (Ok ()) -> None
-        | EAssert assertion ->
-          Format.printf "Assert failure: %a@." Encoding.Expression.pp assertion;
-          let model = get_model solver thread in
-          Format.printf "Model:@.%s@." model;
-          Some thread
-        | ETrap tr ->
-          Format.printf "TRAP: %s@." (Trap.to_string tr);
-          let model = get_model solver thread in
-          Format.printf "Model:@.%s@." model;
-          Some thread
-        | EVal (Error e) ->
-          Format.eprintf "%s@." e;
-          exit 1 )
+        Format.printf "%a@." Expr.pp_list pc;
+        let model = get_model solver pc in
+        let result =
+          match result with
+          | Choice_monad_intf.EVal (Ok ()) -> None
+          | EAssert assertion ->
+            Format.printf "Assert failure: %a@." Expr.pp assertion;
+            Format.printf "Model:@.%a@." Encoding.Model.pp model;
+            Some pc
+          | ETrap tr ->
+            Format.printf "TRAP: %s@." (Trap.to_string tr);
+            Format.printf "Model:@.%a@." Encoding.Model.pp model;
+            Some pc
+          | EVal (Error e) ->
+            Format.eprintf "%s@." e;
+            exit 1
+        in
+        let testcase =
+          List.sort
+            (fun (x1, _) (x2, _) -> compare x1 x2)
+            (Encoding.Model.get_bindings model)
+          >>| snd
```

Comment:
Could we replace this by `|> List.map snd` to avoid defining a custom operator that is used only once ?